### ISSUE TITLE
Crashed on Google Colab

### ISSUE THREAD

**Issue by tkc285** [taichi-dev/taichi#4964]
Hello,

Thank you very much to develop and support this amazing code.

Best regards,

TKC

**Describe the bug**
Successfully install Taichi on Colab without any problems for both version 0.9.0 and 1.0.1. 
Colab Rutime type = "GPU"
Taichi crashed for both versions.

**To Reproduce**
**1. run the fem99 example.**
! ti example fem99 

[Taichi] version 0.9.0, llvm 10.0.0, commit 0c432a37, linux, python 3.7.13
...
Running example fem99 ...
[Taichi] Starting on arch=cuda
[E 05/11/22 20:32:27.539 1427] [x11.cpp:create_window@154] Taichi fails to create a window. This is probably due to the lack of an X11 GUI environment. Consider using the `ti.GUI(show_gui=False)` option, see https://docs.taichi.graphics/lang/articles/misc/gui

***********************************
* Taichi Compiler Stack Traceback *
***********************************
/usr/local/lib/python3.7/dist-packages/taichi/_lib/core/taichi_core.so: taichi::Logger::error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)
/usr/local/lib/python3.7/dist-packages/taichi/_lib/core/taichi_core.so: taichi::GUI::create_window()

**2. 4 line py code**
import taichi as ti
ti.init(arch=ti.gpu)
window = ti.ui.Window('Window Title', (640, 360))
window.show()

[Taichi] version 0.9.0, llvm 10.0.0, commit 0c432a37, linux, python 3.7.13
A error message window " Your session crashed for an unknown reason.  [View runtime logs]"
runtime log 

May 11, 2022, 5:03:09 PM | WARNING | WARNING:root:kernel e8308250-6a34-40e5-aa25-18ef42a35cc7 restarted
-- | -- | --
May 11, 2022, 5:03:09 PM | INFO | KernelRestarter: restarting kernel (1/5), keep random ports
May 11, 2022, 5:03:08 PM | WARNING | cannot initialize GLFW
May 11, 2022, 5:03:07 PM | WARNING | [I 05/11/22 21:03:07.650 1449] [shell.py:_shell_pop_print@33] Graphical python shell detected, using wrapped sys.stdout
May 11, 2022, 5:03:05 PM | INFO | Adapting to protocol v5.1 for kernel e8308250-6a34-40e5-aa25-18ef42a35cc

**Additional comments**
CPU = 11th Gen Intel(R) Core(TM) i7-1195G7 @ 2.90GHz
Ubuntu 20.04.4,    32 GB memory  
video card: 00:02.0 VGA compatible controller: Intel Corporation Device 9a49 (rev 03)

**Comment by strongoier** [comment]
Hi @tkc285. Thanks for reporting this! Just as the error message shows, there is no GUI environment, which is common on servers. You can check out X11 forwarding if you need the visualization.

BTW could you run `ti example minimization` to verify if taichi programs without GUI can work properly? Thanks!

**Comment by strongoier** [comment]
> BTW. the "ti example minimization" works fine. (loss = e-10 level in less than a second)

Thanks for the info!

> if you know how to do it for Colab, Could you please share?

Sorry that I don't have experience about that. There was a related issue #4390, but I don't know whether that can directly apply to Colab. Maybe you can try something like https://stackoverflow.com/questions/60921979/x11-forwarding-through-google-colab.
